session.go

```diff
@@ -74,10 +74,42 @@ type RemoteCallbacks struct {
 	// It is the implementer's responsibility to ensure that the application
 	// recreates the session.
 	OnRestart func(ctx context.Context, sess Session) error
+}
+
+func (cb RemoteCallbacks) HandleStop(ctx context.Context, sess Session) error {
+	if cb.OnStop != nil {
+		return cb.OnStop(ctx, sess)
+	}
+
+	return errors.New("stop callback not set")
+}
+
+func (cb RemoteCallbacks) HandleRestart(ctx context.Context, sess Session) error {
+	if cb.OnRestart != nil {
+		return cb.OnRestart(ctx, sess)
+	}
+
+	return errors.New("restart callback not set")
+}
+
+type RemoteCallbackHandler interface {
+	// Called when a stop is requested via the dashboard or API.
```

Comment:
Nit, idiomatically should be documented as "HandleStop is called ...."

Same for the other godoc comments